src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IForEachLoopStatement.cs

```diff
@@ -872,7 +872,14 @@ public static void M((int, int)[] x)
 IForEachLoopStatement (LoopKind.ForEach) (OperationKind.LoopStatement) (Syntax: 'foreach (va ... }')
   Locals: Local_1: System.Int32 a
     Local_2: System.Int32 b
-  LoopControlVariable: IOperation:  (OperationKind.None) (Syntax: 'var (a, b)')
+  LoopControlVariable: IDeconstructionAssignmentExpression (OperationKind.DeconstructionAssignmentExpression, Type: (System.Int32 a, System.Int32 b)) (Syntax: 'var (a, b)')
+      Left: ITupleExpression (OperationKind.TupleExpression, Type: (System.Int32 a, System.Int32 b)) (Syntax: 'var (a, b)')
```

Comment:
Question for @jcouv @AlekseyTs - Current change exposes the DeconstructionAssignmentExpression within the BoundForEachDeconstructLeft, which ends up with an `IDeconstructionAssignmentExpression` with a left and right. The right part seems redundant to me as the `IConversionExpression` wrapping the `PlaceHolderExpression` is not very useful given that we already expose the conversion on top of the loop collection expression via `IForEachLoopStatement.Collection`, see the Collection dump right below.

Reply:
I feel we should probably only expose the `ITupleExpression` as the `LoopControlVariable` instead of `IDeconstructionAssignmentExpression`. Thoughts?

Also tagging @DustinCampbell

Reply:
How would someone find the `Deconstruct(...)` method that the compiler called?

Reply:
That is tracked with https://github.com/dotnet/roslyn/issues/22477

Reply:
Sure, but wouldn't it be a significant breaking change if we had to switch it *back* to `IDeconstructionAssignmentExpression` later as part of #22477?

Reply:
Ah, get your point. We can keep it in the current form then.